service/src/main/java/com/derpgroup/quip/manager/QuipManager.java

```diff
@@ -34,76 +35,60 @@ public QuipManager(){
     super();
   }
   
-  protected Insults doInsultRequest(Map<String, String> messageMap, SsmlDocumentBuilder builder, QuipMetadata metadata) {
+  protected Quip getRandomQuip(QuipType quipType, Queue<String> recentlyUsedQuips){
+    QuipStore quipStore = QuipStore.getInstance();
+    LOG.debug("Recently used "+quipType.toString().toLowerCase()+"s in user session:\n"+recentlyUsedQuips.toString());
+    int maxQuipHistorySize = determineMaxQuipHistorySize(quipStore.getQuips(quipType).size());
+    Quip quip = quipStore.getRandomQuip(quipType);
+    while(recentlyUsedQuips.contains(quip.getQuipGroup())){
```

Comment:
Maybe we should set up a max number of iterations before we give up? I know it shouldn't ever happen, but just to be safe...